emote/nn/layers.py

```diff
@@ -174,3 +175,159 @@ def get_encoder_output_size(self):
             out_size = np.prod(out_size)
 
         return out_size
+
+
+class FullyConnectedEncoder(nn.Module):
+    def __init__(
+        self,
+        input_size: int,
+        output_size: int,
+        device: torch.device,
+        condition_size: int = 0,
+        hidden_sizes: list[int] = None,
+        freeze_bn: bool = True,
+    ):
+        super().__init__()
+        self.device = device
+
+        self.input_size = input_size
+        self.output_size = output_size
+        self._freeze_bn = freeze_bn
+
+        num_layers = len(hidden_sizes)
+        batch_norm = (
+            FrozenBatchNorm1d(hidden_sizes[0])
+            if self._freeze_bn
+            else nn.BatchNorm1d(hidden_sizes[0])
+        )
+        layers = [
+            nn.Sequential(
+                nn.Linear(input_size + condition_size, hidden_sizes[0]),
+                batch_norm,
+                nn.ReLU(),
+            )
+        ]
+        for i in range(num_layers - 1):
+            batch_norm = (
+                FrozenBatchNorm1d(hidden_sizes[i + 1])
+                if self._freeze_bn
+                else nn.BatchNorm1d(hidden_sizes[i + 1])
+            )
+            layers.append(
+                nn.Sequential(
+                    nn.Linear(hidden_sizes[i], hidden_sizes[i + 1]),
+                    batch_norm,
+                    nn.ReLU(),
+                )
+            )
+        self.hidden_layers = nn.Sequential(*layers).to(self.device)
+        self.output_mean = nn.Linear(hidden_sizes[num_layers - 1], output_size).to(
+            self.device
+        )
+        self.output_log_std = nn.Linear(hidden_sizes[num_layers - 1], output_size).to(
+            self.device
+        )
+
+    def forward(
+        self, data: torch.Tensor, condition: torch.Tensor = None
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        """
+        Running encoder
+
+            Arguments:
+                data (torch.Tensor): batch x data_size
+                condition (torch.Tensor): batch x condition_size
+
+            Returns:
+                tuple(torch.Tensor, torch.Tensor): the mean (batch x data_size)
+                    and log_std (batch x data_size)
+        """
+        assert len(data.shape) == 2
+        if condition is not None:
+            assert len(condition.shape) == 2
+            data = torch.cat((data, condition), dim=1)
```

Comment:
This seems like a such a specific thing... Could we not just compose this with a wrapper instead?